packages/server/src/requestPipeline.ts

```diff
@@ -534,10 +530,16 @@ export async function processGraphQLRequest<TContext extends BaseContext>(
               ),
         ),
       ),
-      http,
     };
 
-    return sendResponse(response);
+    updateResponseHTTP(http);
+
+    if (!requestContext.response.http.statusCode) {
+      // TODO(AS4): This is a bit weird because it can
```

Comment:
Dangling comment

Reply:
I think I made it less weird... was going to say that a few of the things that set it seemed more 500y, but as noted in the PR desc I made those into 500s.